File: JavaToKotlin/app/src/main/java/io/dustin/apps/common/model/extension/NoticeExtension.java
package io.dustin.apps.common.model.extension;

import com.querydsl.core.BooleanBuilder;
import com.querydsl.core.annotations.QueryDelegate;
import io.dustin.apps.admin.domain.notice.model.Notice;
import io.dustin.apps.admin.domain.notice.model.QNotice;

public class NoticeExtension {
    @QueryDelegate(Notice.class)
    public static BooleanBuilder indexByCountPagination(QNotice notice, Long nextId) {
        BooleanBuilder builder = new BooleanBuilder();
        if(nextId != null) {
            builder.and(notice.id.lt(nextId));
        }
        return builder;
    }
}

File: apps/src/main/kotlin/io/dustin/apps/common/model/extension/NoticeExtension.kt
package io.dustin.apps.common.model.extension

//import com.querydsl.core.BooleanBuilder
//import com.querydsl.core.annotations.QueryDelegate
//import io.dustin.apps.admin.domain.notice.model.Notice
//
//object NoticeExtension {
//    @QueryDelegate(Notice::class)
//    fun indexByCountPagination(notice: QNotice, nextId: Long?): BooleanBuilder {
//        val builder = BooleanBuilder()
//        if (nextId != null) {
//            builder.and(notice.id.lt(nextId))
//        }
//        return builder
//    }
//}
